enhancements/update/accepted-risks.md

```diff
@@ -0,0 +1,521 @@
+  `clusterversion.status.conditionalUpdates.risks`.
+- adds a new field 'clusterversion.status.conditionalUpdateRisks.conditions': It
+  contains the observations of the conditional update risk's current status.
+
+For example,
+
+```console
+$ oc clusterversion version -o yaml
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  name: version
+spec:
+  channel: candidate-4.18
+  clusterID: 1c182977-5663-428d-92a3-3d2bdf3fffb6
+  desiredUpdate:
+    accepted:
+    - DualStackNeedsController
+    - LeakedMachineConfigBlocksMCO
+    - OldBootImagesPodmanMissingAuthFlag
+    force: false
+    version: 4.18.15
+status:
+  conditionalUpdateRisks:  # include every risk in the conditional updates (moved up and renamed)
+  - name: DualStackNeedsController
+    message: Upgrade can get stuck on clusters that use multiple networks together with dual stack.
+    url: https://issues.redhat.com/browse/SDN-3996
+    matchingRules:
+    - type: Always
+    conditions:
+    - status: True  # Apply=True if the risk is applied to the current cluster
+      type: Applies
+      reason: MatchingRule
+      message: The matchingRules[0] matches
+      lastTransitionTime: 2021-09-13T17:03:05Z
+  - name: LeakedMachineConfigBlocksMCO
+    message: MCO update might be blocked if some condition holds
+    url: https://issues.redhat.com/browse/SDN-2212
+    matchingRules:
+    - type: PromQL
+      promql:
+        promql: prom_query
+    conditions:
+    - status: False
+      type: Applies
+      reason: NoMatchingRule
+      message: None of the matching rules matches
+      lastTransitionTime: 2021-09-13T17:03:05Z
+  - name: DualStackNeedsController
+    ...
+  - name: RHELKernelHighLoadIOWait
+    ...
+  conditionalUpdates:
+  - release:
+      version: 4.18.16
+      image: quay.io/openshift-release-dev/ocp-release@sha256:abc123
+    riskNames:
+    - DualStackNeedsController
+    - OldBootImagesPodmanMissingAuthFlag
+    - RHELKernelHighLoadIOWait
+    risks:  # deprecated by riskNames
+  - release:
+      version: 4.18.17
+    riskNames:
+    - OldBootImagesPodmanMissingAuthFlag
+    risks:  # deprecated by riskNames
+  - release:
+      version: 4.19.1
+    riskNames:
+    - LeakedMachineConfigBlocksMCO
+    risks:  # deprecated by riskNames
+```
+
+When a conditional update is accepted, the names of its associated risks are going to be merged into `clusterversion.status.history.acceptedRisks` which is an existing field before this enhancement. For example, CVO's acceptance of `4.18.17` leads to `OldBootImagesPodmanMissingAuthFlag` being a part of value of `clusterversion.status.history.acceptedRisks`: The wording might look like "The target release ... is exposed to the risks [OldBootImagesPodmanMissingAuthFlag] and accepted by CVO because all of them are considered acceptable".
+
+When a conditional update is blocked, there is a condition in `clusterversion.status.conditions` with `ReleaseAccepted=False`, e.g.,
```

Comment:
OK so this answers one of my earlier comments, blocking through "ReleaseAccepted=False`  sounds  reasonable.

Reply:
Yeah, a version is either in the `status.conditionalUpdates` or `status.availableUpdates` depending on whether the cluster is exposed to a risk (the former) on that edge or not (the latter).

These status fields are populated by CVO (it gets the data from OSUS, feeds the condition promql (if risk is present) through the cluster's monitoring stack which results in placing the edge to one of these.

Reply:
I'm wondering, if a request to update the desired version did include with it, a list of risks then, whether we could implement, either through CEL on the CRD itself, or through a VAP, a way for it to check the same thing `oc adm upgrade` is doing, but through admission time. Which then would allow `oc edit` or GitOps clients to get the same sort of functionality.

Something to investigate perhaps?

Reply:
Yes, apologies, I meant validating admission policy.

And yes, this was just a comment for a general UX optimisation, it just seemed relevant to this thread as an alternative